skfda/preprocessing/dim_reduction/projection/_fpca.py

```diff
@@ -1,33 +1,445 @@
+        pca_ (sklearn.decomposition.PCA): object for PCA.
+            In both cases (discretized FPCA and basis FPCA) the problem can be
+            reduced to a regular PCA problem and use the framework provided by
+            sklearn to continue.
+
+    Parameters:
+        n_components (int): number of principal components to obtain from
+            functional principal component analysis. Defaults to 3.
+        centering (bool): if True then calculate the mean of the functional data
+            object and center the data first. Defaults to True. If True the
+            passed FDataBasis object is modified.
+        components_basis (Basis): the basis in which we want the principal
+            components. We can use a different basis than the basis contained in
+            the passed FDataBasis object.
+        regularization_lfd (LinearDifferentialOperator, list or int): Linear
+                differential operator. If it is not a LinearDifferentialOperator
+                object, it will be converted to one. If you input an integer
+                then the derivative of that degree will be used to regularize
+                the principal components.
+
+    Examples:
+        Construct an artificial FDataBasis object and run FPCA with this object.
+        The resulting principal components are not compared because there are
+        several equivalent possibilities.
+
+        >>> data_matrix = np.array([[1.0, 0.0], [0.0, 2.0]])
+        >>> sample_points = [0, 1]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> basis = skfda.representation.basis.Monomial((0,1), n_basis=2)
+        >>> basis_fd = fd.to_basis(basis)
+        >>> fpca_basis = FPCABasis(2)
+        >>> fpca_basis = fpca_basis.fit(basis_fd)
+
+    """
+
+    def __init__(self,
+                 n_components=3,
+                 components_basis=None,
+                 centering=True,
+                 regularization_parameter=0,
+                 regularization_lfd=2):
+        """FPCABasis constructor
+
+        Args:
+            n_components (int): number of principal components to obtain from
+                functional principal component analysis
+            components_basis (skfda.representation.Basis): the basis in which we
+                want the principal components. Defaults to None. If so, the
+                basis contained in the passed FDataBasis object for the fit
+                function will be used.
+            centering (bool): if True then calculate the mean of the functional
+                data object and center the data first. Defaults to True
+            regularization_parameter (float): this parameter sets the degree of
+                regularization that is desired. Defaults to 0 (no
+                regularization). When this value is large, the resulting
+                principal components tends to be constant.
+            regularization_lfd (LinearDifferentialOperator, list or int): Linear
```

Comment:
We should try to use the same nomenclature whenever possible in `BasisSmoother`, `MultivariateLinearRegression` and here.